simul/platform/deterlab.go

```diff
@@ -108,71 +119,60 @@ func (d *Deterlab) Configure(pc *Config) {
 	d.sshDeter = make(chan string)
 }
 
+type pkg struct {
+	name      string
+	processor string
+	system    string
+	path      string
+}
+
 // Build prepares all binaries for the Deterlab-simulation.
 // If 'build' is empty, all binaries are created, else only
 // the ones indicated. Either "simul" or "users"
 func (d *Deterlab) Build(build string, arg ...string) error {
-	log.Lvl1("Building for", d.Login, d.Host, d.Project, build, "cothorityDir=", d.cothorityDir)
+	log.Lvl1("Building for", d.Login, d.Host, d.Project, build, "simulDir=", d.simulDir)
 	start := time.Now()
 
 	var wg sync.WaitGroup
 
-	// Start with a clean build-directory
-	current, _ := os.Getwd()
-	log.Lvl3("Current dir is:", current, d.deterDir)
-	defer func() {
-		if err := os.Chdir(current); err != nil {
-			log.Error("Couldn't change back to", current)
-		}
-	}()
-
-	// Go into deterlab-dir and create the build-dir
-	if err := os.Chdir(d.deterDir); err != nil {
-		return err
-	}
 	if err := os.RemoveAll(d.buildDir); err != nil {
 		return err
 	}
 	if err := os.Mkdir(d.buildDir, 0777); err != nil {
 		return err
 	}
 
-	// start building the necessary packages
-	packages := []string{"simul", "users"}
-	if build != "" {
-		packages = strings.Split(build, ",")
+	// start building the necessary binaries - it's always the same,
+	// but built for another architecture.
+	packages := []pkg{
+		{"simul", "amd64", "linux", d.simulDir},
+		{"users", "386", "freebsd", path.Join(d.platformDir, "deterlab_users", ".")},
```

Comment:
for deterlab we need two binaries:
- users - for users.deterlab.net
- simul (former cothority) - for the servers